entrypoint.sh

```diff
@@ -35,9 +30,29 @@ if ! APTIBLE_OUTPUT_FORMAT=json aptible apps | jq -e ".[] | select(.handle == \"
   exit 1
 fi
 
-aptible deploy --environment "$INPUT_ENVIRONMENT" \
-               --app "$INPUT_APP" \
-               --docker-image "$INPUT_DOCKER_IMG" \
-               --private-registry-username "$INPUT_PRIVATE_REGISTRY_USERNAME" \
-               --private-registry-password "$INPUT_PRIVATE_REGISTRY_PASSWORD" \
-               ${INPUT_CONFIG_VARIABLES}
+if [ "$INPUT_TYPE" == "git" ]; then
+  export ACCESS_TOKEN=$(cat "$HOME/.aptible/tokens.json" | jq '.["https://auth.aptible.com"]' -r)
+  REMOTE_URL="root@$INPUT_GIT_REMOTE:$INPUT_ENVIRONMENT/$INPUT_APP.git"
+  git remote add aptible ${REMOTE_URL}
+  BRANCH="deploy-$(date "+%s")"
+  GIT_SSH_COMMAND="ssh -o SendEnv=ACCESS_TOKEN -o PubkeyAuthentication=no -o StrictHostKeyChecking=no -p 43022" git push aptible "main:$BRANCH"
+
+  aptible deploy --environment "$INPUT_ENVIRONMENT" \
+                 --app "$INPUT_APP" \
+                 --git-commitish "$BRANCH" \
+                 ${INPUT_CONFIG_VARIABLES}
```

Comment:
I'm not sure this will work in all cases e.g. config values that contain a space but we do have a disclaimer about using config variables so I'm OK with that for now. If we get to the point of being able to use the API directly, this problem will likely go away.

Reply:
This is how it works currently so I think we can leave it for now.